app/controllers/users_controller.rb

```diff
@@ -21,6 +29,29 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+    return if @user
+    flash[:warning] = t :show
+    redirect_to root_path
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user&.update_attributes user_params
+      flash[:success] = t :update
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @user.destroy
+    flash[:success] = t :delete
+    redirect_to users_url
+  end
```

Comment:
```ruby
if @user.destroy
  flash[:success] = t ".success"
else
  flash[:danger] = t "static_pages.home.usernotfound"
end
redirect_to users_path
```

Reply:
I used:
def destroy 
  @user = User.find_by id: params[:id]
  if @user
    flash[:warning] = "can not destroy"
    redirect_to users_path
  else
    @user.destroy
    flash[:danger] = "deleted account"
  end
  redirect_to users_path
end 
Why it alway return @user = nil and can't destroy  ?

Reply:
1. First let us discuss about your previous code
```ruby
def destroy
  @user.destroy
  flash[:success] = t :delete
  redirect_to users_url
end
```
Here before performing destroy operation, you didn't get the user from the database by it's id. 
That's why @user is nil and don't perform destroy operation. 
2. Secondly, let us discuss about your code that you write on the comment
```ruby
def destroy
  @user = User.find_by id: params[:id]
  if @user
    flash[:warning] = "can not destroy"
    redirect_to users_path
  else
    @user.destroy
    flash[:danger] = "deleted account"
  end
  redirect_to users_path
end
```
-> Here you are getting the user from the database by it's id, it's ok and now this time @user is not nil. 
-> But if you read carefully, you can see that, if user is not nil, that mean the user is present, then
you are not deleting, you are just showing the flash message and redirecting the user to it's users_path. That's why it is not destroying a user.